### PR TITLE
basic url hash functionality to make examples linkable

### PR DESCRIPTION
**Pull request by SoonDead**
This is an attempt to do #30 .

Functionality is preserved. Back button works as expected.

I had to do an unlucky hack because args.name is modified in every call, so I had to copy the whole argument object before feeding it to open.
I did it with `Object.create()`, which is ES5 standard and solves this with simple prototype inheritance.
Every non-crappy browser supports it, but if IE8 support is necessary, I might need to change it.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -749,38 +518,44 @@
...
+        var hash = window.location.hash.substr(1),
+            linkId = 'open-' + hash,
+            example = examples[hash];
+            
+        
+        
+        if (example) {
+            explainGit.reset();
+            document.getElementById(linkId).classList.add('selected');
+            explainGit.open(Object.create(example));
```
Comment (onlywei) [inline index.html]: Nice, I certainly don't care about IE8 at all!

File: index.html [hunk 2]
```diff
@@ -749,38 +518,44 @@
...
-                {id: 'e137e9b', tags: ['master']}
-            ],
-            initialMessage:
-                'Have fun.'
-        });
     };
 
-    openSwitch.addEventListener('click', open, false);
+    window.addEventListener("hashchange", open, false);
+    window.addEventListener("load", open, false);
```
Comment (onlywei) [inline index.html]: It's been a while, but I think I've had every string only use single quotes in this entire codebase?

File: index.html [hunk 3]
```diff
@@ -749,38 +518,44 @@
...
+    
+    function open() {
+        var hash = window.location.hash.substr(1),
+            linkId = 'open-' + hash,
+            example = examples[hash];
+            
+
```
Comment (onlywei) [inline index.html]: Whoa, 3 empty lines.

**Comment by SoonDead** [comment]
I'm going to fix these issues you mentioned in this pull request.

File: js/explaingit.js [hunk 4]
```diff
@@ -5,8 +5,9 @@ define(['historyview', 'controlbox', 'd3'], function (HistoryView, ControlBox, d
         reset,
         explainGit;
 
-    open = function (args) {
-        var name = prefix + args.name,
+    open = function (_args) {
+        var args = Object.create(_args),
```
Comment (SoonDead) [inline js/explaingit.js]: Moved `Object.create()` into `explainGit.open()`.

Renaming the parameter was harder to screw up then changing all internal references.

File: index.html [hunk 5]
```diff
@@ -749,38 +518,42 @@
...
-                {id: 'e137e9b', tags: ['master']}
-            ],
-            initialMessage:
-                'Have fun.'
-        });
     };
 
-    openSwitch.addEventListener('click', open, false);
+    window.addEventListener('hashchange', open, false);
+    window.addEventListener('load', open, false);
```
Comment (SoonDead) [inline index.html]: Single quotes.

File: index.html [hunk 6]
```diff
@@ -749,38 +518,42 @@
...
+    
+    function open() {
+        var hash = window.location.hash.substr(1),
+            linkId = 'open-' + hash,
+            example = examples[hash];
+
```
Comment (SoonDead) [inline index.html]: Only 1 empty line.

**Pull request merged**